pkg/apis/networking/validation/validation_test.go

```diff
@@ -1405,6 +1422,23 @@ func TestValidateIngressUpdate(t *testing.T) {
 			},
 			expectedErrs: field.ErrorList{},
 		},
+		"new Backend.Resource not allowed on update": {
+			tweakIngresses: func(newIngress, oldIngress *networking.Ingress) {
+				oldIngress.Spec.Backend = &defaultBackend
+				newIngress.Spec.Backend = &networking.IngressBackend{
+					Resource: resourceBackend}
+			},
+			expectedErrs: field.ErrorList{field.Forbidden(field.NewPath("spec.backend.resource"), "cannot be set on this object")},
+		},
+		"old Backend.Resource allowed on update": {
+			tweakIngresses: func(newIngress, oldIngress *networking.Ingress) {
+				oldIngress.Spec.Backend = &networking.IngressBackend{
+					Resource: resourceBackend}
+				newIngress.Spec.Backend = &networking.IngressBackend{
+					Resource: resourceBackend}
+			},
+			expectedErrs: field.ErrorList{},
+		},
```

Comment:
additional unit tests:

allowed:
* changing spec.backend from resource -> no resource
* changing a path backend from resource -> no resource
* changing a path backend from resource -> resource
* changing a path backend from no resource -> no resource

not allowed:
* changing a path backend from no resource -> resource